### PR TITLE
fix: add audit filter and update the default AuditCleanInterval

### PR DESCRIPTION
**Pull request by Malyue**
#### What this PR does / why we need it:
add audit filter and update the default AuditCleanInterval

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=532537&iterationID=-1&type=REQUIREMENT)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   add audit filter and update the default AuditCleanInterval           |
| 🇨🇳 中文    |   增加审计日志过滤条件和修改组织及系统日志的默认清理时间           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/core/legacy/endpoints/audit.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func getPermissionBody(listReq *apistructs.AuditsListRequest, identityInfo apist
 		pcr.Scope = apistructs.SysScope
 	} else {
 		pcr.Scope = apistructs.OrgScope
-		pcr.ScopeID = uint64(listReq.OrgID)
+		pcr.ScopeID = listReq.OrgID[0]
```
Comment (sfwn) [inline internal/core/legacy/endpoints/audit.go:310]: I think you should through an error if multiple orgID provided in org scope.

File: internal/core/legacy/model/audit.go [hunk 2]
```diff
@@ -48,6 +48,35 @@ type AuditSettings struct {
 	Config OrgConfig
 }
 
+type ListAuditParam struct {
+	// +required 查询事件起始时间
```
Comment (sfwn) [inline internal/core/legacy/model/audit.go:52]: translate to english, or just ignore it if simple enough.

File: internal/core/legacy/services/audit/audit.go [hunk 3]
```diff
@@ -113,7 +115,113 @@ func (a *Audit) BatchCreateAudit(reqs []apistructs.Audit) error {
...
+		}
+		reqParam.OrgId = []uint64{param.OrgID[0]}
+		if len(param.ProjectID) > 0 {
+			// check if the projectId is owned to the org
+			projectIds, err := a.GetAllProjectIdInOrg(param.OrgID[0])
+			if err != nil {
+				return 0, nil, err
+			}
+			if _, flag := arrays.IsArrayContained(projectIds, param.ProjectID); !flag {
+				return 0, nil, errors.New("用户所选择的项目不属于其所属于的组织")
```
Comment (sfwn) [inline internal/core/legacy/services/audit/audit.go:148]: use i18n.

File: internal/core/legacy/services/audit/audit.go [hunk 4]
```diff
@@ -113,7 +115,113 @@ func (a *Audit) BatchCreateAudit(reqs []apistructs.Audit) error {
...
+		ClientIp:     param.ClientIP,
+		ScopeType:    param.ScopeType,
+	}
+
+	if param.Sys {
+		// if scope_type is sys，there is no need to perform parameter validation
+		reqParam.ProjectID = param.ProjectID
+		reqParam.AppID = param.AppID
+		reqParam.OrgId = param.OrgID
+	} else {
```
Comment (sfwn) [inline internal/core/legacy/services/audit/audit.go:135]: Do not use else here. You can return inside sys code branch directly.
And you can extract non-sys scope logic to a function, and add unit-test for it.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/core/legacy/model/audit.go [hunk 5]
```diff
@@ -48,6 +48,35 @@ type AuditSettings struct {
...
+	// +optional UserID列表
+	UserID []string
+	// +optional FDP项目ID
+	FDPProjectID []string
+	// +optional ScopeID列表
+	ScopeID []uint64
+	// +optional 应用ID列表
+	AppID []uint64
+	// +optional 项目ID列表
+	ProjectID []uint64
```
Comment (sfwn) [inline internal/core/legacy/model/audit.go:69]: test multi-lines comment

File: internal/core/legacy/conf/conf.go [hunk 6]
```diff
@@ -71,11 +71,11 @@ type Conf struct {
...
-	AuditArchiveCron             string `env:"AUDIT_ARCHIVE_CRON" default:"0 0 4 * * ?"`      // audit archive cron
-	SysAuditCleanInterval        int    `env:"SYS_AUDIT_CLEAN_INTERVAL" default:"-7"`         // sys audit clean interval
-	OrgAuditMaxRetentionDays     uint64 `env:"ORG_AUDIT_MAX_RETENTION_DAYS" default:"180"`    // org level audit max retention days
-	OrgAuditDefaultRetentionDays uint64 `env:"ORG_AUDIT_DEFAULT_RETENTION_DAYS" default:"30"` // org level audit default retention days
+	AuditCleanCron               string `env:"AUDIT_CLEAN_CRON" default:"0 0 3 * * ?"`         // audit soft delete cron
+	AuditArchiveCron             string `env:"AUDIT_ARCHIVE_CRON" default:"0 0 4 * * ?"`       // audit archive cron
+	SysAuditCleanInterval        int    `env:"SYS_AUDIT_CLEAN_INTERVAL" default:"-30"`         // sys audit clean interval
+	OrgAuditMaxRetentionDays     uint64 `env:"ORG_AUDIT_MAX_RETENTION_DAYS" default:"500"`     // org level audit max retention days
+	OrgAuditDefaultRetentionDays uint64 `env:"ORG_AUDIT_DEFAULT_RETENTION_DAYS" default:"365"` // org level audit default retention days
```
Comment (sfwn) [inline internal/core/legacy/conf/conf.go:78]: test multi-lines comment again

File: internal/core/legacy/conf/conf.go [hunk 7]
```diff
@@ -71,11 +71,11 @@ type Conf struct {
...
 
 	// audit
-	AuditCleanCron               string `env:"AUDIT_CLEAN_CRON" default:"0 0 3 * * ?"`        // audit soft delete cron
-	AuditArchiveCron             string `env:"AUDIT_ARCHIVE_CRON" default:"0 0 4 * * ?"`      // audit archive cron
-	SysAuditCleanInterval        int    `env:"SYS_AUDIT_CLEAN_INTERVAL" default:"-7"`         // sys audit clean interval
-	OrgAuditMaxRetentionDays     uint64 `env:"ORG_AUDIT_MAX_RETENTION_DAYS" default:"180"`    // org level audit max retention days
-	OrgAuditDefaultRetentionDays uint64 `env:"ORG_AUDIT_DEFAULT_RETENTION_DAYS" default:"30"` // org level audit default retention days
+	AuditCleanCron               string `env:"AUDIT_CLEAN_CRON" default:"0 0 3 * * ?"`         // audit soft delete cron
+	AuditArchiveCron             string `env:"AUDIT_ARCHIVE_CRON" default:"0 0 4 * * ?"`       // audit archive cron
+	SysAuditCleanInterval        int    `env:"SYS_AUDIT_CLEAN_INTERVAL" default:"-30"`         // sys audit clean interval
```
Comment (sfwn) [inline internal/core/legacy/conf/conf.go:76]: test again

File: internal/core/legacy/dao/application.go [hunk 8]
```diff
@@ -231,3 +231,9 @@ func (client *DBClient) GetApplicationsByNames(projectID uint64, names []string)
 	err := client.Where("project_id = ?", projectID).Where("name in (?)", names).Find(&applications).Error
 	return applications, err
 }
+
+func (client *DBClient) GetApplicationsByOrgId(orgId uint64) ([]model.Application, error) {
```
[the remark this answers is not in the document]
Comment (Malyue) [reply internal/core/legacy/dao/application.go]: has been fixed

File: internal/core/legacy/endpoints/audit.go (hunk 1 above)
Comment (Malyue) [reply internal/core/legacy/endpoints/audit.go:317]: has been fixed

File: internal/core/legacy/model/audit.go (hunk 2 above)
Comment (Malyue) [reply internal/core/legacy/model/audit.go]: has been fixed

File: internal/core/legacy/services/audit/audit.go (hunk 3 above)
Comment (Malyue) [reply internal/core/legacy/services/audit/audit.go]: has been fixed

File: internal/core/legacy/services/audit/audit.go (hunk 4 above)
Comment (Malyue) [reply internal/core/legacy/services/audit/audit.go]: has been fixed

File: cmd/erda-server/conf/i18n/i18n.yaml [hunk 9]
```diff
@@ -21,6 +21,12 @@ zh:
   APPLICATION: 应用
   ErrAddMemberOwner: "添加所有者失败，数量超过上限"
 
+  # audit error
+  ErrInvalidOrg: "OrgID 非法，可能是不存在orgID或者多个orgID"
```
Comment (sfwn) [inline cmd/erda-server/conf/i18n/i18n.yaml:25]: Add spaces between Chinese and English text.
And use same case for orgID.

File: cmd/erda-server/conf/i18n/i18n.yaml [hunk 10]
```diff
@@ -43,3 +49,9 @@ en:
   PROJECT: project
   APPLICATION: application
   ErrAddMemberOwner: "failed to add project owner, quantity exceeds limit"
+
+  # audit error
+  ErrInvalidOrgID: "Invalid OrgID,it may not exist or multiple OrgIDs"
```
Comment (sfwn) [inline cmd/erda-server/conf/i18n/i18n.yaml:54]: Check text style. Add blank if necessary.

File: internal/core/legacy/endpoints/audit.go [hunk 11]
```diff
@@ -109,6 +109,8 @@ func (e *Endpoints) ListAudits(ctx context.Context, r *http.Request, vars map[st
 		return apierrors.ErrListAudit.InvalidParameter(err).ToResp(), nil
 	}
 
+	ctx = context.WithValue(ctx, "lang_codes", i18n.Language(r))
```
Comment (sfwn) [inline internal/core/legacy/endpoints/audit.go:112]: a bit strange.

File: internal/core/legacy/services/audit/audit.go [hunk 12]
```diff
@@ -111,9 +121,152 @@ func (a *Audit) BatchCreateAudit(reqs []apistructs.Audit) error {
...
+	}
+
+	return a.db.GetAuditsByParam(filterParam)
+}
+
+// constructFilterParamByReq valid the param and construct the filterParam to query db by `apistruct.AuditsListRequest`
+func (a *Audit) constructFilterParamByReq(ctx context.Context, param *apistructs.AuditsListRequest) (*model.ListAuditParam, error) {
+	langCodes, ok := ctx.Value("lang_codes").(i18n.LanguageCodes)
+	if !ok {
+		return nil, errors.New("Invalid Language")
```
Comment (sfwn) [inline internal/core/legacy/services/audit/audit.go:161]: missing language

File: internal/core/legacy/services/permission/permission.go [hunk 13]
```diff
@@ -172,6 +173,11 @@ var checkAdaptor security.PermissionAdaptor
 func (p *Permission) CheckPermission(req *apistructs.PermissionCheckRequest) (bool, error) {
 	logrus.Debugf("invoke permission, time: %s, req: %+v", time.Now().Format(time.RFC3339), req)
 
+	// if the param is invalid,it will set the req as nil
```
Comment (sfwn) [inline internal/core/legacy/services/permission/permission.go:176]: add space after `,`.

File: internal/core/legacy/services/permission/permission.go [hunk 14]
```diff
@@ -172,6 +173,11 @@ var checkAdaptor security.PermissionAdaptor
...
+	if req == nil {
+		return false, errors.New("the user operation is invalid,maybe the param is invalid")
```
Comment (sfwn) [inline internal/core/legacy/services/permission/permission.go:178]: same.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: cmd/erda-server/conf/i18n/i18n.yaml (hunk 9 above)
Comment (Malyue) [reply cmd/erda-server/conf/i18n/i18n.yaml]: has been fixed

File: cmd/erda-server/conf/i18n/i18n.yaml (hunk 10 above)
Comment (Malyue) [reply cmd/erda-server/conf/i18n/i18n.yaml]: has been fixed

File: internal/core/legacy/endpoints/audit.go (hunk 11 above)
Comment (Malyue) [reply internal/core/legacy/endpoints/audit.go]: Already encapsulated as a method

File: internal/core/legacy/services/audit/audit.go (hunk 12 above)
Comment (Malyue) [reply internal/core/legacy/services/audit/audit.go]: has been fixed

File: internal/core/legacy/services/permission/permission.go (hunk 13 above)
Comment (Malyue) [reply internal/core/legacy/services/permission/permission.go]: has been fixed

File: internal/core/legacy/services/permission/permission.go (hunk 14 above)
Comment (Malyue) [reply internal/core/legacy/services/permission/permission.go]: has been fixed

File: internal/core/legacy/services/audit/audit_test.go [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package audit
+
+import (
+	"context"
+	"fmt"
+	"github.com/erda-project/erda/internal/core/legacy/common/ctxhelper"
```
Comment (sfwn) [inline internal/core/legacy/services/audit/audit_test.go:20]: adjust imports order

File: internal/core/legacy/services/permission/permission.go [hunk 16]
```diff
@@ -172,6 +173,11 @@ var checkAdaptor security.PermissionAdaptor
 func (p *Permission) CheckPermission(req *apistructs.PermissionCheckRequest) (bool, error) {
 	logrus.Debugf("invoke permission, time: %s, req: %+v", time.Now().Format(time.RFC3339), req)
 
+	// if the param is invalid, it will set the req as nil
+	if req == nil {
+		return false, errors.New("the user operation is invalid, maybe the param is invalid")
```
Comment (sfwn) [inline internal/core/legacy/services/permission/permission.go:178]: make it concise: `empty request`

File: cmd/erda-server/conf/i18n/i18n.yaml [hunk 17]
```diff
@@ -21,6 +21,12 @@ zh:
   APPLICATION: 应用
   ErrAddMemberOwner: "添加所有者失败，数量超过上限"
 
+  # audit error
+  ErrInvalidOrg: "OrgID 非法，可能是不存在OrgID或者多个OrgID"
```
Comment (sfwn) [inline cmd/erda-server/conf/i18n/i18n.yaml:25]: add space between Chinese and English.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/core/legacy/services/permission/permission.go (hunk 16 above)
Comment (Malyue) [reply internal/core/legacy/services/permission/permission.go]: modified the param validation logic of orgID, and the issue has been fixed

**Review by sfwn — APPROVED** [review]

**Pull request merged**